src/sentry/status_checks/slack_integration_version.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+
+import random
+from sentry import features, options
+from sentry.models import OrganizationMember, OrganizationIntegration
+from sentry.utils.compat import map, filter
+from sentry.utils import metrics
+
+from .base import StatusCheck, Problem
+
+
+def is_wst_slack(oi):
+    integration = oi.integration
+    if integration.provider != "slack":
+        return False
+    metadata = integration.metadata
+    # no installation type sent and no user access token
+    return "installation_type" not in metadata and "user_access_token" not in metadata
+
+
+class SlackIntegrationVersion(StatusCheck):
+    def should_run_queries(self):
+        # using sampling here so we can slowly increase the load here
```

Comment:
This could just be checked once and memoized. It won't change until a process restart.